dvc/dvcfile.py

```diff
@@ -234,6 +234,8 @@ def remove_stage(self, stage):
         logger.debug("Removing '%s' from '%s'", stage.name, self.path)
         del d["stages"][stage.name]
         dump_yaml(self.path, d)
+        if len(d["stages"]) == 0:
+            self.remove(force=True)
```

Comment:
Let's not dump if there's going to be no stages at all.

Reply:
Also, I think it's wrong to assume that lock file and dvc.yaml will always have the same no. of entries.

Reply:
> Let's not dump if there's going to be no stages at all.

Fixed

Reply:
> Also, I think it's wrong to assume that lock file and dvc.yaml will always have the same no. of entries. I hope they'd be the same, but at the end, they are two different files (one of which, you're allowed to edit/remove yourselves).

I'm not sure I've got you right but now `dvc.lock` and `dvc.yaml` are removed independently. Please, check it out.